core/services/flux_monitor.go

```diff
@@ -250,30 +257,61 @@ func NewPollingDeviationChecker(
 		precision:    initr.InitiatorParams.Precision,
 		runManager:   runManager,
 		currentPrice: decimal.NewFromInt(0),
+		currentRound: big.NewInt(0),
 		fetcher:      fetcher,
 		delay:        delay,
+		newRounds:    make(chan eth.Log),
 	}, nil
 }
 
 // Initialize retrieves the price that's on-chain, with which we must check
 // the deviation from.
 func (p *PollingDeviationChecker) Initialize(client eth.Client) error {
 	price, err := client.GetAggregatorPrice(p.address, p.precision)
+	if err != nil {
+		return err
+	}
 	p.currentPrice = price
-	return err
+
+	round, err := client.GetAggregatorRound(p.address)
```

Comment:
Ideally the node would check if it had participated in the newest round via [latestSubmission](https://github.com/smartcontractkit/chainlink/blob/develop/evm/v0.5/contracts/dev/PrepaidAggregator.sol#L366). I'll break it out, seems like an edge case that could be it's own story.